### ISSUE TITLE
Master OOME during GCUpgrade9to10TestIT (test fails)

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#1435]
I am seeing repeated failures in GCUpgrade9to10TestIT ever since #1427 . It seems to be caused by the Master server getting an OutOfMemoryError, and the rest of the test waiting for it to come back before it times out. However, I cannot get a consistent stack trace in the logs from the master, to identify where in the code it is failing. I think it's in a child thread, because it keeps triggering the UncaughtExceptionHandler, and printing a basic message to the `.out` file that MiniAccumuloCluster creates, as in:

```
Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Thread-2"
```

The name of the thread is different each time, and it's unfortunate we don't have these threads named better. Jstack is no help, because the process is already dead by the time the I see the problem (which seems to happen quickly). Analyzing the relevant upgrade and test code seems to be the best chance to track this down.

I've also seen this, but not every time:

```
Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "process reaper"
OpenJDK 64-Bit Server VM warning: Exception java.lang.OutOfMemoryError occurred dispatching signal SIGTERM to handler- the VM may need to be forcibly terminated
```

**Comment by jzgithub1** [comment]
@ctubbsii ,  this is the only error I see when running this test  but it is not related to OOME.  The call to deleteLock in killMacGc  always fails. 

`2020-02-11 14:59:33,790 [functional.ConfigurableMacBase] ERROR: Unable to delete ZooLock for mini accumulo-gc
java.lang.IllegalStateException: No lock is held at /accumulo/8d868a12-346d-4e31-98ba-fed5f5e04a16/gc/lock
	at org.apache.accumulo.fate.zookeeper.ZooLock.deleteLock(ZooLock.java:499)
	at org.apache.accumulo.test.upgrade.GCUpgrade9to10TestIT.killMacGc(GCUpgrade9to10TestIT.java:94)`
Maybe we can just remove that try catch block with ZooLock.deleteLock in it and add a check to see if the GC process is actually running before we kill it to keep the intent and purpose of the killMacGC function in tact.  

Here is the code change I suggest:
```java
private void killMacGc() throws ProcessNotFoundException, InterruptedException {
    // kill gc started by MAC
    assertNotNull(getCluster().getProcesses().get(ServerType.GARBAGE_COLLECTOR));
    getCluster().killProcess(ServerType.GARBAGE_COLLECTOR,
        getCluster().getProcesses().get(ServerType.GARBAGE_COLLECTOR).iterator().next());
    assertNull(getCluster().getProcesses().get(ServerType.GARBAGE_COLLECTOR));
  }
```

**Comment by ctubbsii** [comment]
I think this was fixed in #1441 (b0bbff007d372fb3a9e3c4c4f900e063a65a40b1)
